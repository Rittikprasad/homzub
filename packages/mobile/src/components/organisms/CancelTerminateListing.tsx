import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import { Formik, FormikProps, FormikValues } from 'formik';
import { WithTranslation, withTranslation } from 'react-i18next';
import * as yup from 'yup';
import { DateFormats, DateUtils } from '@homzhub/common/src/utils/DateUtils';
import { FormUtils } from '@homzhub/common/src/utils/FormUtils';
import { theme } from '@homzhub/common/src/styles/theme';
import { Label } from '@homzhub/common/src/components/atoms/Text';
import { TextArea } from '@homzhub/common/src/components/atoms/TextArea';
import { FormButton } from '@homzhub/common/src/components/molecules/FormButton';
import { FormCalendar } from '@homzhub/common/src/components/molecules/FormCalendar';
import { FormDropdown, IDropdownOption } from '@homzhub/common/src/components/molecules/FormDropdown';

export interface ISubmitPayload extends IFormData {
  isTerminate: boolean;
  description: string;
}

export interface IFormData {
  reasonId: number;
  terminationDate: string;
}

interface IScreenState {
  formData: IFormData;
  description: string;
  isTouched: boolean;
}

interface IProps {
  isTerminate?: boolean;
  leaseEndDate?: string;
  onFormEdit: () => void;
  reasonData: IDropdownOption[];
  onSubmit: (payload: ISubmitPayload) => void;
}

type Props = IProps & WithTranslation;

class CancelTerminateListing extends Component<Props, IScreenState> {
  constructor(props: Props) {
    super(props);
    this.state = {
      formData: {
        reasonId: 0,
        terminationDate: '',
      },
      description: '',
      isTouched: false,
    };
  }

  public render(): React.ReactNode {
    const { formData, description, isTouched } = this.state;
    const { t, isTerminate = false, onFormEdit, reasonData, leaseEndDate } = this.props;

    return (
      <View style={styles.form}>
        <Formik
          onSubmit={this.handleSubmit}
          initialValues={{ ...formData }}
          validate={FormUtils.validate(this.formSchema)}
        >
          {(formProps: FormikProps<FormikValues>): React.ReactNode => {
            const { reasonId, terminationDate } = formProps.values;
            const isButtonEnable = isTerminate ? reasonId > 0 && !!terminationDate : reasonId > 0;

            if (!isTouched && formProps.dirty) {
              this.setState({ isTouched: true });
              onFormEdit();
            }

            return (
              <>
                <FormDropdown
                  name="reasonId"
                  placeholder={t('common:reasonExample')}
                  label={t('common:reason')}
                  isMandatory
                  options={reasonData}
                  formProps={formProps}
                  dropdownContainerStyle={styles.dropdownStyle}
                />
                {isTerminate && (
                  <FormCalendar
                    formProps={formProps}
                    isCurrentDateEnable
                    maxDate={leaseEndDate && DateUtils.getUtcFormatted(leaseEndDate, DateFormats.DD_MM_YYYY)}
                    name="terminationDate"
                    textType="label"
                    label={t('property:terminationDate')}
                    calendarTitle={t('property:selectMoveOutDate')}
                    placeHolder={t('property:selectMoveOutDate')}
                    isMandatory
                  />
                )}
                <TextArea
                  value={description}
                  placeholder={t('common:typeDescriptionHere')}
                  label={t('assetDescription:description')}
                  wordCountLimit={500}
                  containerStyle={styles.textArea}
                  onMessageChange={this.handleDescription}
                />
                <Label type="large" style={styles.message}>
                  {t('common:actionNotDone')}
                </Label>
                <FormButton
                  // @ts-ignore
                  onPress={formProps.handleSubmit}
                  containerStyle={{ backgroundColor: isButtonEnable ? theme.colors.error : theme.colors.disabled }}
                  formProps={formProps}
                  disabled={!isButtonEnable}
                  type="primary"
                  title={isTerminate ? t('common:sendNotice') : t('property:deleteListing')}
                />
              </>
            );
          }}
        </Formik>
      </View>
    );
  }

  private handleDescription = (value: string): void => {
    this.setState({ description: value });
  };

  private formSchema = (): yup.ObjectSchema<IFormData> => {
    const { t } = this.props;

    return yup.object().shape({
      terminationDate: yup.string().when('isTerminate', {
        is: true,
        then: yup.string().required(t('moreProfile:fieldRequiredError')),
      }),
      reasonId: yup.number().required(t('moreProfile:fieldRequiredError')),
    });
  };

  private handleSubmit = (values: IFormData): void => {
    const { onSubmit, isTerminate } = this.props;
    const { description } = this.state;
    onSubmit({ ...values, isTerminate: isTerminate ?? false, description });
  };
}

export default withTranslation()(CancelTerminateListing);

const styles = StyleSheet.create({
  form: {
    paddingVertical: 14,
  },
  dropdownStyle: {
    paddingVertical: 12,
  },
  textArea: {
    marginTop: 10,
    marginBottom: 50,
  },
  message: {
    alignSelf: 'center',
    color: theme.colors.darkTint5,
    marginVertical: 12,
  },
});
