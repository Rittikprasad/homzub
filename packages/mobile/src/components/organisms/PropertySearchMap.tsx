import React from 'react';
import { StyleSheet } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import MapView, { Marker, PROVIDER_GOOGLE } from 'react-native-maps';
import { PropertyUtils } from '@homzhub/common/src/utils/PropertyUtils';
import { Point } from '@homzhub/common/src/services/GooglePlaces/interfaces';
import { theme } from '@homzhub/common/src/styles/theme';
import { CustomMarker } from '@homzhub/common/src/components/atoms/CustomMarker';
import { SnapCarousel } from '@homzhub/mobile/src/components/atoms/Carousel';
import { PropertyMapCard } from '@homzhub/mobile/src/components/molecules/PropertyMapCard';
import { Asset } from '@homzhub/common/src/domain/models/Asset';
import { Attachment } from '@homzhub/common/src/domain/models/Attachment';
import { IFilter } from '@homzhub/common/src/domain/models/Search';

const {
  viewport: { width },
  DeviceDimensions: { SMALL },
} = theme;
const SLIDER_WIDTH = width <= SMALL.width ? 300 : width * 0.8;
const MAP_DELTA = 0.1;

interface IState {
  currentSlide: number;
}

interface IProps {
  searchLocation: Point;
  properties: Asset[];
  transaction_type: number;
  onSelectedProperty: (item: Asset) => void;
  getPropertiesListView: () => void;
  setFilter: (payload: IFilter) => void;
  filters: IFilter;
}
type Props = IProps & WithTranslation;

export class PropertySearchMap extends React.PureComponent<Props, IState> {
  private mapRef: MapView | null = null;
  private carouselRef = null;

  public state = {
    currentSlide: 0,
  };

  public componentDidUpdate = (prevProps: Props): void => {
    const {
      properties,
      searchLocation,
      filters: { offset },
    } = this.props;
    if (properties.length <= 0) {
      this.animateCamera(searchLocation.lat, searchLocation.lng);
      return;
    }

    if (prevProps.properties.length !== properties.length && offset === 0) {
      this.animateCamera(properties[0].latitude, properties[0].longitude);
      // @ts-ignore
      this.carouselRef.snapToItem(0);
    }
  };

  public render(): React.ReactNode {
    const { currentSlide } = this.state;
    const { properties, searchLocation } = this.props;
    let { lat: initLatitude, lng: initLongitude } = searchLocation;

    if (properties.length > 0) {
      initLatitude = properties[0].latitude;
      initLongitude = properties[0].longitude;
    }

    return (
      <>
        <MapView
          ref={(mapRef): void => {
            this.mapRef = mapRef;
          }}
          provider={PROVIDER_GOOGLE}
          style={styles.mapView}
          initialRegion={{
            latitude: initLatitude,
            longitude: initLongitude,
            latitudeDelta: MAP_DELTA,
            longitudeDelta: MAP_DELTA,
          }}
        >
          {properties.map((property: Asset, index: number) => {
            const { latitude, longitude } = property;
            const onMarkerPress = (): void => {
              this.onMarkerPress(index);
            };
            return (
              <Marker key={`${property.id}${index}`} coordinate={{ latitude, longitude }} onPress={onMarkerPress}>
                <CustomMarker selected={index === currentSlide} />
              </Marker>
            );
          })}
        </MapView>
        <SnapCarousel
          containerStyle={styles.carouselStyle}
          carouselData={properties}
          activeIndex={currentSlide}
          itemWidth={SLIDER_WIDTH}
          carouselItem={this.renderCarouselItem}
          onSnapToItem={this.onSnapToItem}
          onLoadMore={this.loadMoreProperties}
          bubbleRef={(ref): void => {
            this.carouselRef = ref;
          }}
          testID="assetSnap"
        />
      </>
    );
  }

  private renderCarouselItem = (item: Asset): React.ReactElement => {
    const { transaction_type, onSelectedProperty } = this.props;
    const {
      attachments,
      projectName,
      furnishing,
      spaces,
      carpetAreaUnit,
      carpetArea,
      assetGroup: { code },
      leaseTerm,
      saleTerm,
      isAssetOwner,
    } = item;
    const price = this.getPrice(item);
    const amenities = PropertyUtils.getAmenities(spaces, furnishing, code, carpetArea, carpetAreaUnit?.title ?? '');
    const image = attachments.filter((currentImage: Attachment) => currentImage.isCoverImage);
    let currencyData = item.country.currencies[0];

    if (leaseTerm && leaseTerm.currency) {
      currencyData = leaseTerm.currency;
    }

    if (saleTerm && saleTerm.currency) {
      currencyData = saleTerm.currency;
    }

    const navigateToAssetDetails = (): void => {
      if (leaseTerm) {
        onSelectedProperty(item);
      }
      if (saleTerm) {
        onSelectedProperty(item);
      }
    };

    return (
      <PropertyMapCard
        source={image[0]?.link ?? null}
        name={projectName}
        currency={currencyData}
        price={price}
        leaseListingId={leaseTerm?.id}
        saleListingId={saleTerm?.id}
        priceUnit={transaction_type === 0 ? 'mo' : ''}
        amenitiesData={amenities}
        onSelectedProperty={navigateToAssetDetails}
        isAssetOwner={isAssetOwner}
      />
    );
  };

  public onSnapToItem = (currentSlide: number): void => {
    const { properties } = this.props;
    const { latitude, longitude } = properties[currentSlide];

    this.animateCamera(latitude, longitude);
    this.setState({ currentSlide });
  };

  private onMarkerPress = (index: number): void => {
    // @ts-ignore
    this.carouselRef.snapToItem(index);
  };

  private animateCamera = (latitude: number, longitude: number): void => {
    this.mapRef?.animateCamera({
      center: {
        longitude,
        latitude,
      },
    });
  };

  public getPrice = (item: Asset): number => {
    const { leaseTerm, saleTerm } = item;
    if (leaseTerm) {
      return Number(leaseTerm.expectedPrice);
    }
    if (saleTerm) {
      return Number(saleTerm.expectedPrice);
    }
    return 0;
  };

  public loadMoreProperties = (): void => {
    const {
      setFilter,
      getPropertiesListView,
      filters: { offset },
      properties,
    } = this.props;

    setFilter({ offset: (offset ?? 0) + properties.length, is_sorting: false });
    getPropertiesListView();
  };
}

export default withTranslation()(PropertySearchMap);
const styles = StyleSheet.create({
  mapView: {
    flex: 1,
  },
  carouselStyle: {
    paddingBottom: 10,
    position: 'absolute',
    bottom: 14,
  },
});
