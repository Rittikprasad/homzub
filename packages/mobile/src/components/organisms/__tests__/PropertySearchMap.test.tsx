import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import { PropertySearchMap } from '@homzhub/mobile/src/components/organisms/PropertySearchMap';
import { PropertySearchData } from '@homzhub/common/src/mocks/PropertySearchData';
import { mockAsset } from '@homzhub/common/src/mocks/AssetDescription';
import { SearchFilter } from '@homzhub/common/src/mocks/FilterData';

let props: any;
let wrapper: ShallowWrapper;

describe('PropertySearchMap', () => {
  const createTestProps = (testProps: any): object => ({
    searchLocation: {
      lat: 1,
      lng: 1,
    },
    properties: PropertySearchData,
    transaction_type: 1,
    onSelectedProperty: jest.fn(),
    getPropertiesListView: jest.fn(),
    setFilter: jest.fn(),
    filters: SearchFilter,
    ...testProps,
  });

  props = createTestProps({});

  it('should match snapshot', () => {
    wrapper = shallow(<PropertySearchMap {...props} t={(key: string): string => key} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshot for carouselItem', () => {
    wrapper = shallow(<PropertySearchMap {...props} t={(key: string): string => key} />);
    // @ts-ignore
    wrapper.find('[testID="assetSnap"]').prop('carouselItem')(mockAsset);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshot for Snap To Item', () => {
    wrapper = shallow(<PropertySearchMap {...props} t={(key: string): string => key} />);
    // @ts-ignore
    wrapper.find('[testID="assetSnap"]').prop('onSnapToItem')(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshot for carouselItem when transaction_type is 0', () => {
    props = createTestProps({
      transaction_type: 0,
    });
    wrapper = shallow(<PropertySearchMap {...props} t={(key: string): string => key} />);
    // @ts-ignore
    wrapper.find('[testID="assetSnap"]').prop('carouselItem')(mockAsset);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshot when properties are empty', () => {
    props = createTestProps({
      properties: [],
    });
    wrapper = shallow(<PropertySearchMap {...props} t={(key: string): string => key} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
