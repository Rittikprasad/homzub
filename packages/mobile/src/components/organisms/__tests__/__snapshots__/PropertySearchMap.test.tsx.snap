// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertySearchMap should match snapshot 1`] = `
<Fragment>
  <MapView
    initialRegion={
      Object {
        "latitude": "25.61900000",
        "latitudeDelta": 0.1,
        "longitude": "85.12300000",
        "longitudeDelta": 0.1,
      }
    }
    provider="google"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <MapMarker
      coordinate={
        Object {
          "latitude": "25.61900000",
          "longitude": "85.12300000",
        }
      }
      key="10"
      onPress={[Function]}
      stopPropagation={false}
    >
      <Memo(CustomMarker)
        selected={true}
      />
    </MapMarker>
    <MapMarker
      coordinate={
        Object {
          "latitude": "25.31400000",
          "longitude": "85.12350000",
        }
      }
      key="21"
      onPress={[Function]}
      stopPropagation={false}
    >
      <Memo(CustomMarker)
        selected={false}
      />
    </MapMarker>
  </MapView>
  <SnapCarousel
    activeIndex={0}
    bubbleRef={[Function]}
    carouselData={
      Array [
        Object {
          "asset_group": Object {
            "id": 1,
            "name": "Residential",
          },
          "asset_type": Object {
            "id": 2,
            "name": "Apartment / Condo",
          },
          "block_number": "B",
          "carpet_area": "5000",
          "carpet_area_unit": Object {
            "id": 1,
            "label": "Square feet",
            "name": "SQ_FT",
            "title": "Sq.ft",
          },
          "country": Object {
            "currencies": Array [],
          },
          "floor_number": 1,
          "furnishing": "SEMI",
          "id": 1,
          "images": Array [
            Object {
              "file_name": "House-1a",
              "is_cover_image": true,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
          ],
          "latitude": "25.61900000",
          "lease_term": Object {
            "currency_code": "INR",
            "expected_price": "20000.0000",
            "id": 1,
          },
          "longitude": "85.12300000",
          "project_name": "Malti Kunj",
          "spaces": Array [
            Object {
              "count": 3,
              "id": 1,
              "name": "Bedroom",
            },
            Object {
              "count": 3,
              "id": 2,
              "name": "Bathroom",
            },
          ],
          "total_floors": 4,
          "unit_number": "41",
        },
        Object {
          "asset_group": Object {
            "id": 1,
            "name": "Residential",
          },
          "asset_type": Object {
            "id": 2,
            "name": "Apartment / Condo",
          },
          "block_number": "B",
          "carpet_area": "2000",
          "carpet_area_unit": Object {
            "id": 1,
            "label": "Square feet",
            "name": "SQ_FT",
            "title": "Sq.ft",
          },
          "country": Object {
            "currencies": Array [
              Object {
                "currency_code": "",
                "currency_name": "",
                "currency_symbol": "",
              },
            ],
            "id": 1,
            "iso2_code": "",
            "iso3_code": "",
            "name": "",
            "phone_codes": Array [],
          },
          "floor_number": 2,
          "furnishing": "SEMI",
          "id": 2,
          "images": Array [
            Object {
              "file_name": "House-1a",
              "is_cover_image": true,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
            Object {
              "file_name": "House-1b",
              "is_cover_image": true,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
          ],
          "latitude": "25.31400000",
          "lease_term": Object {
            "currency_code": "INR",
            "expected_price": "22000.0000",
            "id": 2,
          },
          "longitude": "85.12350000",
          "project_name": "Mira Sadan",
          "spaces": Array [
            Object {
              "count": 4,
              "id": 1,
              "name": "Bedroom",
            },
            Object {
              "count": 2,
              "id": 2,
              "name": "Bathroom",
            },
          ],
          "total_floors": 3,
          "unit_number": "42",
        },
      ]
    }
    carouselItem={[Function]}
    containerStyle={
      Object {
        "bottom": 14,
        "paddingBottom": 10,
        "position": "absolute",
      }
    }
    itemWidth={600}
    onLoadMore={[Function]}
    onSnapToItem={[Function]}
    testID="assetSnap"
  />
</Fragment>
`;

exports[`PropertySearchMap should match snapshot for Snap To Item 1`] = `
<Fragment>
  <MapView
    initialRegion={
      Object {
        "latitude": "25.61900000",
        "latitudeDelta": 0.1,
        "longitude": "85.12300000",
        "longitudeDelta": 0.1,
      }
    }
    provider="google"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <MapMarker
      coordinate={
        Object {
          "latitude": "25.61900000",
          "longitude": "85.12300000",
        }
      }
      key="10"
      onPress={[Function]}
      stopPropagation={false}
    >
      <Memo(CustomMarker)
        selected={false}
      />
    </MapMarker>
    <MapMarker
      coordinate={
        Object {
          "latitude": "25.31400000",
          "longitude": "85.12350000",
        }
      }
      key="21"
      onPress={[Function]}
      stopPropagation={false}
    >
      <Memo(CustomMarker)
        selected={true}
      />
    </MapMarker>
  </MapView>
  <SnapCarousel
    activeIndex={1}
    bubbleRef={[Function]}
    carouselData={
      Array [
        Object {
          "asset_group": Object {
            "id": 1,
            "name": "Residential",
          },
          "asset_type": Object {
            "id": 2,
            "name": "Apartment / Condo",
          },
          "block_number": "B",
          "carpet_area": "5000",
          "carpet_area_unit": Object {
            "id": 1,
            "label": "Square feet",
            "name": "SQ_FT",
            "title": "Sq.ft",
          },
          "country": Object {
            "currencies": Array [],
          },
          "floor_number": 1,
          "furnishing": "SEMI",
          "id": 1,
          "images": Array [
            Object {
              "file_name": "House-1a",
              "is_cover_image": true,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
          ],
          "latitude": "25.61900000",
          "lease_term": Object {
            "currency_code": "INR",
            "expected_price": "20000.0000",
            "id": 1,
          },
          "longitude": "85.12300000",
          "project_name": "Malti Kunj",
          "spaces": Array [
            Object {
              "count": 3,
              "id": 1,
              "name": "Bedroom",
            },
            Object {
              "count": 3,
              "id": 2,
              "name": "Bathroom",
            },
          ],
          "total_floors": 4,
          "unit_number": "41",
        },
        Object {
          "asset_group": Object {
            "id": 1,
            "name": "Residential",
          },
          "asset_type": Object {
            "id": 2,
            "name": "Apartment / Condo",
          },
          "block_number": "B",
          "carpet_area": "2000",
          "carpet_area_unit": Object {
            "id": 1,
            "label": "Square feet",
            "name": "SQ_FT",
            "title": "Sq.ft",
          },
          "country": Object {
            "currencies": Array [
              Object {
                "currency_code": "",
                "currency_name": "",
                "currency_symbol": "",
              },
            ],
            "id": 1,
            "iso2_code": "",
            "iso3_code": "",
            "name": "",
            "phone_codes": Array [],
          },
          "floor_number": 2,
          "furnishing": "SEMI",
          "id": 2,
          "images": Array [
            Object {
              "file_name": "House-1a",
              "is_cover_image": true,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
            Object {
              "file_name": "House-1b",
              "is_cover_image": true,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
          ],
          "latitude": "25.31400000",
          "lease_term": Object {
            "currency_code": "INR",
            "expected_price": "22000.0000",
            "id": 2,
          },
          "longitude": "85.12350000",
          "project_name": "Mira Sadan",
          "spaces": Array [
            Object {
              "count": 4,
              "id": 1,
              "name": "Bedroom",
            },
            Object {
              "count": 2,
              "id": 2,
              "name": "Bathroom",
            },
          ],
          "total_floors": 3,
          "unit_number": "42",
        },
      ]
    }
    carouselItem={[Function]}
    containerStyle={
      Object {
        "bottom": 14,
        "paddingBottom": 10,
        "position": "absolute",
      }
    }
    itemWidth={600}
    onLoadMore={[Function]}
    onSnapToItem={[Function]}
    testID="assetSnap"
  />
</Fragment>
`;

exports[`PropertySearchMap should match snapshot for carouselItem 1`] = `
<Fragment>
  <MapView
    initialRegion={
      Object {
        "latitude": "25.61900000",
        "latitudeDelta": 0.1,
        "longitude": "85.12300000",
        "longitudeDelta": 0.1,
      }
    }
    provider="google"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <MapMarker
      coordinate={
        Object {
          "latitude": "25.61900000",
          "longitude": "85.12300000",
        }
      }
      key="10"
      onPress={[Function]}
      stopPropagation={false}
    >
      <Memo(CustomMarker)
        selected={true}
      />
    </MapMarker>
    <MapMarker
      coordinate={
        Object {
          "latitude": "25.31400000",
          "longitude": "85.12350000",
        }
      }
      key="21"
      onPress={[Function]}
      stopPropagation={false}
    >
      <Memo(CustomMarker)
        selected={false}
      />
    </MapMarker>
  </MapView>
  <SnapCarousel
    activeIndex={0}
    bubbleRef={[Function]}
    carouselData={
      Array [
        Object {
          "asset_group": Object {
            "id": 1,
            "name": "Residential",
          },
          "asset_type": Object {
            "id": 2,
            "name": "Apartment / Condo",
          },
          "block_number": "B",
          "carpet_area": "5000",
          "carpet_area_unit": Object {
            "id": 1,
            "label": "Square feet",
            "name": "SQ_FT",
            "title": "Sq.ft",
          },
          "country": Object {
            "currencies": Array [],
          },
          "floor_number": 1,
          "furnishing": "SEMI",
          "id": 1,
          "images": Array [
            Object {
              "file_name": "House-1a",
              "is_cover_image": true,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
          ],
          "latitude": "25.61900000",
          "lease_term": Object {
            "currency_code": "INR",
            "expected_price": "20000.0000",
            "id": 1,
          },
          "longitude": "85.12300000",
          "project_name": "Malti Kunj",
          "spaces": Array [
            Object {
              "count": 3,
              "id": 1,
              "name": "Bedroom",
            },
            Object {
              "count": 3,
              "id": 2,
              "name": "Bathroom",
            },
          ],
          "total_floors": 4,
          "unit_number": "41",
        },
        Object {
          "asset_group": Object {
            "id": 1,
            "name": "Residential",
          },
          "asset_type": Object {
            "id": 2,
            "name": "Apartment / Condo",
          },
          "block_number": "B",
          "carpet_area": "2000",
          "carpet_area_unit": Object {
            "id": 1,
            "label": "Square feet",
            "name": "SQ_FT",
            "title": "Sq.ft",
          },
          "country": Object {
            "currencies": Array [
              Object {
                "currency_code": "",
                "currency_name": "",
                "currency_symbol": "",
              },
            ],
            "id": 1,
            "iso2_code": "",
            "iso3_code": "",
            "name": "",
            "phone_codes": Array [],
          },
          "floor_number": 2,
          "furnishing": "SEMI",
          "id": 2,
          "images": Array [
            Object {
              "file_name": "House-1a",
              "is_cover_image": true,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
            Object {
              "file_name": "House-1b",
              "is_cover_image": true,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
          ],
          "latitude": "25.31400000",
          "lease_term": Object {
            "currency_code": "INR",
            "expected_price": "22000.0000",
            "id": 2,
          },
          "longitude": "85.12350000",
          "project_name": "Mira Sadan",
          "spaces": Array [
            Object {
              "count": 4,
              "id": 1,
              "name": "Bedroom",
            },
            Object {
              "count": 2,
              "id": 2,
              "name": "Bathroom",
            },
          ],
          "total_floors": 3,
          "unit_number": "42",
        },
      ]
    }
    carouselItem={[Function]}
    containerStyle={
      Object {
        "bottom": 14,
        "paddingBottom": 10,
        "position": "absolute",
      }
    }
    itemWidth={600}
    onLoadMore={[Function]}
    onSnapToItem={[Function]}
    testID="assetSnap"
  />
</Fragment>
`;

exports[`PropertySearchMap should match snapshot for carouselItem when transaction_type is 0 1`] = `
<Fragment>
  <MapView
    initialRegion={
      Object {
        "latitude": "25.61900000",
        "latitudeDelta": 0.1,
        "longitude": "85.12300000",
        "longitudeDelta": 0.1,
      }
    }
    provider="google"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <MapMarker
      coordinate={
        Object {
          "latitude": "25.61900000",
          "longitude": "85.12300000",
        }
      }
      key="10"
      onPress={[Function]}
      stopPropagation={false}
    >
      <Memo(CustomMarker)
        selected={true}
      />
    </MapMarker>
    <MapMarker
      coordinate={
        Object {
          "latitude": "25.31400000",
          "longitude": "85.12350000",
        }
      }
      key="21"
      onPress={[Function]}
      stopPropagation={false}
    >
      <Memo(CustomMarker)
        selected={false}
      />
    </MapMarker>
  </MapView>
  <SnapCarousel
    activeIndex={0}
    bubbleRef={[Function]}
    carouselData={
      Array [
        Object {
          "asset_group": Object {
            "id": 1,
            "name": "Residential",
          },
          "asset_type": Object {
            "id": 2,
            "name": "Apartment / Condo",
          },
          "block_number": "B",
          "carpet_area": "5000",
          "carpet_area_unit": Object {
            "id": 1,
            "label": "Square feet",
            "name": "SQ_FT",
            "title": "Sq.ft",
          },
          "country": Object {
            "currencies": Array [],
          },
          "floor_number": 1,
          "furnishing": "SEMI",
          "id": 1,
          "images": Array [
            Object {
              "file_name": "House-1a",
              "is_cover_image": true,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
            Object {
              "file_name": "House-1b",
              "is_cover_image": false,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
          ],
          "latitude": "25.61900000",
          "lease_term": Object {
            "currency_code": "INR",
            "expected_price": "20000.0000",
            "id": 1,
          },
          "longitude": "85.12300000",
          "project_name": "Malti Kunj",
          "spaces": Array [
            Object {
              "count": 3,
              "id": 1,
              "name": "Bedroom",
            },
            Object {
              "count": 3,
              "id": 2,
              "name": "Bathroom",
            },
          ],
          "total_floors": 4,
          "unit_number": "41",
        },
        Object {
          "asset_group": Object {
            "id": 1,
            "name": "Residential",
          },
          "asset_type": Object {
            "id": 2,
            "name": "Apartment / Condo",
          },
          "block_number": "B",
          "carpet_area": "2000",
          "carpet_area_unit": Object {
            "id": 1,
            "label": "Square feet",
            "name": "SQ_FT",
            "title": "Sq.ft",
          },
          "country": Object {
            "currencies": Array [
              Object {
                "currency_code": "",
                "currency_name": "",
                "currency_symbol": "",
              },
            ],
            "id": 1,
            "iso2_code": "",
            "iso3_code": "",
            "name": "",
            "phone_codes": Array [],
          },
          "floor_number": 2,
          "furnishing": "SEMI",
          "id": 2,
          "images": Array [
            Object {
              "file_name": "House-1a",
              "is_cover_image": true,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
            Object {
              "file_name": "House-1b",
              "is_cover_image": true,
              "link": "https://www.investopedia.com/thmb/7GOsX_NmY3KrIYoZPWOu6SldNFI=/735x0/houses_and_land-5bfc3326c9e77c0051812eb3.jpg",
            },
          ],
          "latitude": "25.31400000",
          "lease_term": Object {
            "currency_code": "INR",
            "expected_price": "22000.0000",
            "id": 2,
          },
          "longitude": "85.12350000",
          "project_name": "Mira Sadan",
          "spaces": Array [
            Object {
              "count": 4,
              "id": 1,
              "name": "Bedroom",
            },
            Object {
              "count": 2,
              "id": 2,
              "name": "Bathroom",
            },
          ],
          "total_floors": 3,
          "unit_number": "42",
        },
      ]
    }
    carouselItem={[Function]}
    containerStyle={
      Object {
        "bottom": 14,
        "paddingBottom": 10,
        "position": "absolute",
      }
    }
    itemWidth={600}
    onLoadMore={[Function]}
    onSnapToItem={[Function]}
    testID="assetSnap"
  />
</Fragment>
`;

exports[`PropertySearchMap should match snapshot when properties are empty 1`] = `
<Fragment>
  <MapView
    initialRegion={
      Object {
        "latitude": 1,
        "latitudeDelta": 0.1,
        "longitude": 1,
        "longitudeDelta": 0.1,
      }
    }
    provider="google"
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <SnapCarousel
    activeIndex={0}
    bubbleRef={[Function]}
    carouselData={Array []}
    carouselItem={[Function]}
    containerStyle={
      Object {
        "bottom": 14,
        "paddingBottom": 10,
        "position": "absolute",
      }
    }
    itemWidth={600}
    onLoadMore={[Function]}
    onSnapToItem={[Function]}
    testID="assetSnap"
  />
</Fragment>
`;
